### ISSUE TITLE
File Block: file uploads fail with a generic error for file types I don't have permissions to upload and then the post cannot be published

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/gutenberg-mobile#2910]
Steps to reproduce:

1. Open a WordPress.com Free site (no plan) in the app.
2. Go to My Site > Blog Posts > + > Blog post.
3. Add a file block and choose an mp3 file from the device ([or any file type that requires a plan](https://wordpress.com/support/accepted-filetypes/)).
4. Observe that the block now shows "( ! ) Error" in red text.
5. Observe that if you leave and re-open the post the error goes away (but the upload is still in a failed state).
6. Observe that the file is labeled "Upload Failed" in the Media screen.
7. Re-open the post.
8. Try to preview the post.
9. Try to publish the post

**Result:** if I try to upload a file type that is not allowed for my plan (i.e. an mp3 file on a free site) to a file block, the upload will fail with a generic error and then I cannot preview or publish the post. ([2m59s](https://cloudup.com/cQ5KQoUxRb2))

"( ! ) Error" in red text|No error after close/reopen|"Upload Failed" in Media
---|---|---
![Screen Shot 2020-12-16 at Dec 16 4 05 46 PM](https://user-images.githubusercontent.com/1119271/102417501-12b0ac00-3fb9-11eb-827c-8d7bb57a6b5d.png)|![Screen Shot 2020-12-16 at Dec 16 4 06 00 PM](https://user-images.githubusercontent.com/1119271/102417538-1f350480-3fb9-11eb-95aa-db09818d2300.png)|![Screen Shot 2020-12-16 at Dec 16 4 06 09 PM](https://user-images.githubusercontent.com/1119271/102417591-34119800-3fb9-11eb-8c1d-ff5ec9a94c53.png)
Saving is stuck on preview or publish|"We couldn't upload this media, and didn't publish the post." shows in Posts List
![Screen Shot 2020-12-16 at Dec 16 4 06 40 PM](https://user-images.githubusercontent.com/1119271/102417620-4095f080-3fb9-11eb-8287-93d19e3dbf41.png)|![Screen Shot 2020-12-16 at Dec 16 4 07 42 PM](https://user-images.githubusercontent.com/1119271/102417647-53102a00-3fb9-11eb-94d2-2c3d99ef94c6.png)

<sup>Tested with WPAndroid 16.4-rc-1 on Pixel 3 Android 11 on WordPress.com Free site flibbitygibbit.home.blog logged in as user907.</sup>

**Comment by jd-alexander** [comment]
> Result: if I try to upload a file type that is not allowed for my plan (i.e. an mp3 file on a free site) to a file block, the upload will fail with a generic error 

Thanks for reporting @designsimply in this case would it have been better if an error was shown pertaining to the fact that your plan doesn't support file uploads of that type? 

> and then I cannot preview or publish the post. 

I think this is the expected behavior because, in theory, the upload failed so the post contains a File Block that has failed content attached to it. So to preview or publish the post you would need to fix the file by either removing it or replacing it. Let me know what you think :)

**Comment by designsimply** [comment]
Yes, it would be awesome if the error could say something like, "You do not have permission to upload this file type. Please check which file types are allowed by your plan." … or just the first sentence if that works better, but my wish (if possible!) would be to have the error be really specific in this case because it's a clear problem we can probably detect.

Okay, if it's the expected behavior that you cannot preview or publish a post with a failed upload in it, then the error should definitely be persistent. In my test, I found that the error disappears after closing and re-opening the post.

**Comment by jd-alexander** [comment]
> I found that the error disappears after closing and re-opening the post.

I will check this out. The error behavior here is tied to the behavior that has existed within the app for other media blocks such as Image & Video. I will verify this and report here with my findings.

**Comment by jd-alexander** [comment]
> Saving is stuck on preview or publish

This error seems to be present for other media blocks as well. Once an upload has failed and the preview button is pressed then the "Saving..." dialog doesn't get dismissed. 

> "Upload Failed" in Media

This seems to be the expected behavior here.

**Comment by guarani** [comment]
Just removing the milestone because we normally only add them to PRs, not issues. Let me know if that's incorrect and we can revert.

**Comment by guarani** [comment]
Just checked with @jd-alexander and this issue is Android-specific, so added the android label above.

**Comment by SiobhyB** [comment]
Going ahead to remove the `[Pri] High` label. Although the error messaging still needs to be improved and is currently confusing, it doesn't lead to content loss and the fact we've had no additional reports for over two years indicates to me that the priority isn't high. Please feel free to add the label and comment if you disagree with that assessment around priority!
